crates/cargo-contract/src/cmd/build/mod.rs

```diff
@@ -650,7 +658,26 @@ pub(crate) fn execute(args: ExecuteArgs) -> Result<BuildResult> {
             &crate_metadata.contract_artifact_name,
         )?;
 
-        Ok(optimization_result)
+        let cargo_contract_version = if let Ok(version) = Version::parse(VERSION) {
+            version
+        } else {
+            anyhow::bail!(
+                "Unable to parse version number for the currently running \
+                    `cargo-contract` binary."
+            );
+        };
+
+        let build_info = BuildInfo {
+            rustc_version: crate::util::rustc_toolchain()?,
```

Comment:
This field should be renamed `rust_toolchain` or so, as the actual version of `rustc`is in the `compiler` field/